pkg/appender/ingest.go

```diff
@@ -241,6 +241,9 @@ func (mc *MetricsCache) postMetricUpdates(metric *MetricState) {
 		if !sent {
 			if metric.store.samplesQueueLength() == 0 {
 				metric.setState(storeStateReady)
+				if metric.store.numNotProcessed == 0 && metric.store.pending.Len() == 0 {
```

Comment:
`metric.store.pending.Len() == 0` is the same as `metric.store.samplesQueueLength() == 0`